library/cartridge_instance.py

```diff
@@ -14,21 +15,115 @@
 }
 
 
+def serialize_dict(d):
+    parts = []
+
+    for k, v in d.items():
+        value_str = None
+        if type(v) == str:
+            value_str = "'%s'" % v
+        elif type(v) == float or type(v) == int:
+            value_str = "%s" % v
+        elif type(v) == bool:
+            value_str = 'true' if v else 'false'
+        else:
+            raise CartridgeException(
+                cartridge_errcodes.BAD_VALUE_TYPE,
+                "Unknown value type: {}" % type(v)
+            )
+
+        parts.append("%s = %s" % (k, value_str))
+
+    return '{ %s }' % ', '.join(parts)
+
+
+def box_cfg_was_called(control_console):
+    return control_console.eval('''
+        return type(box.cfg) ~= 'function'
+    ''')
+
+
+def get_box_cfg(control_console):
+    return control_console.eval('''
+        return box.cfg or box.NULL
+    ''')
+
+
+def change_memory_size(param_name, cartridge_defaults, config, control_console):
```

Comment:
I'm not sure you need handle memory change separately. It should be handled as others dynamic options. Yes, it can't be decreased but you need just rethrow box error

Reply:
The main difference between memory changing and other dynamic params is that memory size can be only increased in runtime (but it still can be decreased by instance restart), but all other dynamic params can be changed in runtime. So

* if memory changing fails, there are two cases:
  * "cannot be decreased error" -> it's impossible to change memory in runtime, need to restart
  * all other errors -> fail, something went wrong
* if any other parameter changing fails there is only one case:
  * fail, something went wrong